frontend/public/components/utils/managed-by.tsx

```diff
@@ -0,0 +1,94 @@
+import * as React from 'react';
+import * as classNames from 'classnames';
+import { Link } from 'react-router-dom';
+
+import { ResourceIcon } from './resource-icon';
+import {
+  groupVersionFor,
+  K8sResourceCommon,
+  K8sResourceKind,
+  referenceForGroupVersionKind,
+  referenceForModel,
+  referenceForOwnerRef,
+  OwnerReference,
+} from '../../module/k8s';
+import { ClusterServiceVersionModel } from '@console/operator-lifecycle-manager/src/models';
+import { ClusterServiceVersionKind } from '@console/operator-lifecycle-manager';
+import { k8sList } from '../../module/k8s/resource';
+
+const isOwnedByOperator = (csv: K8sResourceKind, owner: OwnerReference) => {
+  const { version } = groupVersionFor(owner.apiVersion);
+  return csv.spec.customresourcedefinitions.owned.some(
+    (api) => api.kind === owner.kind && version === api.version,
+  );
+};
+
+const matchOwnerAndCSV = (owner: OwnerReference, csvs: K8sResourceCommon[] = []) => {
+  return csvs.find((csv) => isOwnedByOperator(csv, owner));
+};
+
+const ManagedByOperatorResourceLink: React.SFC<ManagerLinkProps> = (props) => {
+  const { csvName, namespace, owner } = props;
+  const name = owner.name;
+  const { group, version } = groupVersionFor(owner.apiVersion);
+  const kind = owner.kind;
+
+  const reference = referenceForGroupVersionKind(group)(version)(kind);
+
+  const link = `/k8s/ns/${namespace}/${referenceForModel(
+    ClusterServiceVersionModel,
+  )}/${csvName}/${reference}/${name}`;
+
+  return (
+    <span className="co-resource-item">
+      <ResourceIcon kind={referenceForOwnerRef(owner)} />
+      <Link to={link} className="co-resource-item__resource-name" data-test-operand-link={name}>
+        {name}
+      </Link>
+    </span>
+  );
+};
+
+export const ManagedByOperatorLink: React.SFC<ManagedByLinkProps> = (props) => {
+  const { obj, className } = props;
+  const [data, setData] = React.useState<ClusterServiceVersionKind[]>();
+  const namespace = obj.metadata.namespace;
+  React.useEffect(() => {
+    k8sList(ClusterServiceVersionModel, { ns: namespace })
+      .then(setData)
+      .catch(
+        // eslint-disable-next-line no-console
+        console.error('Could not fetch CSVs'),
+      );
+  }, [namespace]);
+  const ownerReferences = obj?.metadata?.ownerReferences || [];
```

Comment:
You technically don't need the `?` since it would have blown up above with `obj.metadata.namespace`. We should either always have the `?` after `obj` or not.

```suggestion
  const ownerReferences = obj.metadata.ownerReferences || [];
```